api/src/Model/Model/IndexedModel.php

```diff
@@ -70,15 +79,25 @@ public function countIndex()
 
     public function getIndexDocumentData()
     {
-        $modelArray = $this->toArray();
-
-        foreach ($modelArray as $attribute => &$value) {
-            if ($value instanceof Carbon) {
-                $value = $value->toIso8601String();
+        $relations = [];
+
+        if (! empty($this->indexRelations)) {
+            // We have to do this because we don't know if the relation is one to or one to many. If it is one to one
+            // we don't want to strip out the keys.
```

Comment:
We will need to fix this datetime casting in future as now the datetimes will be serialised as a Carbon instance, not iso8601 string